[DOCS] Reformats cluster health and cluster state APIs (#45206)
elastic/elasticsearch: docs/reference/cluster/health.asciidoc

controlled by the worst shard status. The cluster status is controlled by the 
worst index status.

One of the main benefits of the API is the ability to wait until the cluster 
reaches a certain high water-mark health level. For example, the following will 
wait for 50 seconds for the cluster to reach the `yellow` level (if it reaches 
the `green` or `yellow` status before 50 seconds elapse, it will return at that 
point):

[source,js]
--------------------------------------------------
GET /_cluster/health?wait_for_status=yellow&timeout=50s
--------------------------------------------------
// CONSOLE

[[cluster-health-api-path-params]]
==== {api-path-parms-title}

include::{docdir}/rest-api/common-parms.asciidoc[tag=index]

[[cluster-health-api-query-params]]
==== {api-query-parms-title}

`level`::
    (Optional, string) Can be one of `cluster`, `indices` or `shards`. Controls 
    the details level of the health information returned. Defaults to `cluster`.
    
include::{docdir}/rest-api/common-parms.asciidoc[tag=local]
    
include::{docdir}/rest-api/common-parms.asciidoc[tag=timeoutparms]

`wait_for_active_shards`::

>     A number controlling to how many active shards to wait for, `all` to wait

Reviewer: Needs a `(Optional, string)` indicator.

Reviewer: Several of the other indicators in this section also need these indicators.

`number_of_data_nodes`::
    (integer) The number of nodes that are dedicated data nodes.

`active_primary_shards`::
    (integer) The number of active primary shards.

`active_shards`::
    (integer) The total number of active primary and replica shards.

`relocating_shards`::
    (integer) The number of shards that are under relocation.

`initializing_shards`::
    (integer) The number of shards that are under initialization.

`unassigned_shards`::
    (integer) The number of shards that are not allocated.

`delayed_unassigned_shards`::
    (integer) The number of shards whose allocation has been delayed by the 
    timeout settings.

`number_of_pending_tasks`::
    (integer) The number of cluster-level changes that have not yet been 
    executed.

`number_of_in_flight_fetch`::
    (integer) The number of unfinished fetches.

`task_max_waiting_in_queue_millis`::
    (integer) The time expressed in milliseconds since the earliest initiated task 
    is waiting for being performed.

`active_shards_percent_as_number`::
    (float) The ratio of active shards in the cluster expressed as a percentage. 

[[cluster-health-api-example]]
==== {api-examples-title}

> 

Reviewer: Adding this request should clear up the elasticsearch-ci/docs-check and elasticsearch-ci/oss-distro-docs tests.
```suggestion

[source,js]
--------------------------------------------------
GET _cluster/health
--------------------------------------------------
// CONSOLE
// TEST[s/^/PUT test1\n/]

```

[[cluster-health]]
=== Cluster Health

> 

Reviewer: Per the [API Reference template](https://raw.githubusercontent.com/elastic/docs/master/shared/api-ref-ex.asciidoc), we'll typically use a short description here.
```suggestion
Returns the health status of a cluster
```

[[cluster-health-api-request]]
==== {api-request-title}

> `GET _cluster/health` +

Reviewer: I think this should be something like
` GET _cluster/health/{index}`

I don't think we need a separate example for comma-separated values.

`GET /_cluster/health/test1,test2`

[[cluster-health-api-desc]]
==== {api-description-title}

> The cluster health API allows to get a simple status on the health of the 

Reviewer: ```suggestion
The cluster health API returns a simple status on the health of the 
```

cluster. The API can also be executed against one or more indices to get just 
the specified indices health.

The cluster health status is: `green`, `yellow` or `red`. On the shard level, a 
`red` status indicates that the specific shard is not allocated in the cluster, 
`yellow` means that the primary shard is allocated but replicas are not, and 
`green` means that all shards are allocated. The index level status is 
controlled by the worst shard status. The cluster status is controlled by the 
worst index status.

One of the main benefits of the API is the ability to wait until the cluster 
reaches a certain high water-mark health level. For example, the following will 
wait for 50 seconds for the cluster to reach the `yellow` level (if it reaches 
the `green` or `yellow` status before 50 seconds elapse, it will return at that 
point):

[source,js]
--------------------------------------------------
GET /_cluster/health?wait_for_status=yellow&timeout=50s
--------------------------------------------------
// CONSOLE

> [[cluster-health-api-response-body]]

Reviewer: We need a  path parameters section following the desciption.

I would also move the response body **after** the query parameters.

==== {api-response-body-title}

> The API returns the following response in case of a quiet single node cluster 

Reviewer: I believe the response body title should be parameter definitions.

You can move the example snippet to the `[[cluster-health-api-example]]` section.

Author: I added the param definitions to the section and moved the snippet to the examples. Please have a look at the param definitions because these were not provided. I created them, but some might be inexact.

Author: I'm not sure whether we should list the params in alphabetical order or rather in the same order as they are presented in the response body. Please let me know what do you prefer.

Reviewer: Feel free to leave this section out for now.

If you decide to include it, let's go with the same order as the response body. Thanks @szabosteve!
